neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -122,6 +122,26 @@ public UInt160[] GetScriptHashesForVerifying(Snapshot snapshot)
             return hashes.OrderBy(p => p).ToArray();
         }
 
+        public virtual bool Reverify(Snapshot snapshot, IEnumerable<Transaction> mempool)
+        {
+            if (ValidUntilBlock <= snapshot.Height || ValidUntilBlock > snapshot.Height + MaxValidUntilBlockIncrement)
+                return false;
+            if (NativeContract.Policy.GetBlockedAccounts(snapshot).Intersect(GetScriptHashesForVerifying(snapshot)).Count() > 0)
+                return false;
+            BigInteger balance = NativeContract.GAS.BalanceOf(snapshot, Sender);
+            BigInteger fee = SystemFee + NetworkFee;
+            if (balance < fee) return false;
+            fee += mempool.Where(p => p != this && p.Sender.Equals(Sender)).Select(p => (BigInteger)(p.SystemFee + p.NetworkFee)).Sum();
```

Comment:
@erikzhang, just to be sure, this line is checking the fee of all transactions from this `Sender`, right?
If the sum exceeds balance than the current `tx` become invalid, yes?

Reply:
we could try to unit test this part... looks like it works, but this ref comparison may be uncertain indeed.

Reply:
I mean the logic, I believe it is correct.

But UT would be nice also.

Reply:
Lets UT on a next PR, and unlock this one.

Reply:
ok, but there is just one pending major, why clean mempool and not reverify even with higher fees?